SOC/common/simulation.py

```diff
@@ -230,24 +268,34 @@ def open(self, file_name = 'sim'):
     
 
     def get_exponent(self,
-                     column='AvalancheSize',
+                     column: str = 'AvalancheSize',
                      low: int = 1,
                      high: int = 10,
-                     plot = True, 
-                     plot_filename=None):
+                     plot: bool = True,
+                     plot_filename: typing.Optional[str] = None) -> dict:
+        """
+        Plot histogram of gathered data from data_df,
+
+        :param column: which column of data_df should be visualized?
+        :type column: str
+        :param low: lower cutoff for log-log-linear fit
+        :type low: int
+        :param high: higher cutoff for log-log-linear fit
+        :type high: int
+        :param plot: if False, skips all plotting and just returns fit parameters
+        :type plot: bool
+        :param plot_filename: optional filename for saved plot. This skips displaying the plot!
+        :type plot_filename: bool
+        :return: fit parameters
+        :rtype: dict
+        """
         df = self.data_df
         filtered = df.loc[df.number_of_iterations != 0, column]
         sizes, counts = np.unique(filtered, return_counts=True)
-        filtered = df.number_of_iterations[df.number_of_iterations != 0]
-        sizes, counts = np.unique(filtered, return_counts=True)
         indices = (low < sizes) & (sizes < high)
-        coef_a, coef_b = poly = np.polyfit(np.log10(sizes[indices]), np.log10(counts[indices]), 1)
-        # assert smooth_width < hist_num
-        #     second_deriv = grab_second_deriv(log_heights, smooth_width)
-        #     ind_min, ind_max = find_largest_true_block(np.abs(second_deriv) <= d2_cutoff)
-        # else:
-        #     ind_min, ind_max = cutoffs
-
+        coef_a, coef_b = poly = np.polyfit(np.log10(sizes[indices]),
+                                           np.log10(counts[indices]),
+                                           1)
         if plot:
             fig, ax = plt.subplots()
             ax.loglog(sizes, counts, "o", label="Data")
```

Comment:
```suggestion
            ax.loglog(sizes, counts, "o", ms=1, label="data")
```

Reply:
W czwartej linijce niżej (sorry, że tu wpisuję te mini poprawki):
```suggestion
                      label=fr"$y = {10**coef_b:.1f}\ \exp({coef_a:.4f} x)$", alpha=0.5)
```

Reply:
Spoczko, jest git :)

Reply:
Właśnie zobaczyłem co faktycznie zmieniasz; przezroczystość zawsze na propsie

Reply:
Podmieniłem markersize=1, bo to straszne hardkodowanie, na '.' zamiast 'o'. IMO będzie git.